### DISCUSSION TITLE
Why does removeConsole SWC plugin run in development

### DISCUSSION THREAD

**Discussion by ndejoma** [vercel/next.js#42568] [Ideas]
This issue has been bugging me regarding the new`removeConsole` swc compiler option. If enabled, the plugin should only run in the build phase as it `does not make sense to remove the `mighty` console.log in development. In my opinion, I think it should only run in build phases and not in development where developers use `console.log` to debug their code and I cannot add a console statement for it to be removed as write the code in development.

Performing this check is not necessary, current I am using Next  `v12.3.1`, so I am not sure whether the behavior has changed from Next `v13`. I had to modify the config to check if `phase === PHASE_PRODUCTION_BUILD`.

```js
   const nextConfig = function (phase, { defaultConfig }) {
	return {
               ...restOfMyConfig
		compiler: {
			removeConsole: phase === PHASE_PRODUCTION_BUILD ? true : false
		}
	};
};
```

**Comment by icyJoseph** [comment]
Yeah, well, people could argue the other way around too. I want to make sure that the console is indeed being removed even while I am developing. I've seen many asking how to do X thing from Prod in Dev, and vice-versa. It is hard to land a sensible default that satisfies everyone.

What's the most sensible default here is a bit up in the air. At least you've got the option to control it via phases. 

The default behaviour doesn't remove them AFAIK, perhaps that's the most sensible default? Because once you decide to activate it, then you are in charge.

In Next 13 it is still the same I believe.

**Reply by ndejoma** [reply]
Why would you want to remove console statements while developing, yet you are the one who added them for debugging purposes, that was my issue. Thank you @icyJoseph for sharing your view on this issue. It still bugs me, but I am gonna have to stick with checking the `phase`.

**Reply by icyJoseph** [reply]
Like I said, people would take any position on this matter to get out of it. I could say, well console logging to debug is not very good either. People use a tools like Pino to transport the data you need to log to a logging layer. If you want to find issues use debugging tools, etc. 

I think the most sensible thing to do is just as you have it, but perhaps you don't need the full ternary:

```js
const nextConfig = function (phase, { defaultConfig }) {
	return {
               ...restOfMyConfig
		compiler: {
			removeConsole: phase === PHASE_PRODUCTION_BUILD
		}
	};
};
```
